samples/HttpRequestFeaturesv2/Controllers/PersonalizerController.cs

```diff
@@ -0,0 +1,176 @@
+
+        public PersonalizerController(PersonalizerClient personalizerClient)
+        {
+            this.client = personalizerClient;
+        }
+
+        /// <summary>
+        /// Creates a RankRequest with user time of day, HTTP request features,
+        /// and taste as the context and several different foods as the actions
+        /// </summary>
+        /// <returns>RankRequest with user info</returns>
+        [HttpGet("GenerateRank")]
+        public RankRequest GenerateRank()
+        {
+            string eventId = Guid.NewGuid().ToString();
+
+            // Get the actions list to choose from personalizer with their features.
+            IList<RankableAction> actions = GetActions();
+
+            // Get context information from the user.
+            HttpRequestFeatures httpRequestFeatures = GetHttpRequestFeaturesFromRequest(Request);
+            string timeOfDayFeature = GetUsersTimeOfDay();
+            string tasteFeature = GetUsersTastePreference();
+
+            // Create current context from user specified data.
+            IList<object> currentContext = new List<object>() {
+                    new { time = timeOfDayFeature },
+                    new { taste = tasteFeature },
+                    new { httpRequestFeatures }
+            };
+
+            // Exclude an action for personalizer ranking. This action will be held at its current position.
+            IList<string> excludeActions = new List<string> { "juice" };
+
+            // Rank the actions
+            return new RankRequest(actions, currentContext, excludeActions, eventId);
+        }
+
+        /// <summary>
+        /// Sends a rank to the personalizer client
+        /// </summary>
+        /// <param name="rankRequest"></param>
+        /// <returns>RankResponse from request</returns>
+        [HttpPost("PostRank")]
+        public RankResponse PostRank([FromBody]RankRequest rankRequest)
+        {
+            return client.Rank(rankRequest);
+        }
+
+        /// <summary>
+        /// Creates a RewardRequest with value of 0
+        /// </summary>
+        /// <returns>RewardRequest with value of 0</returns>
+        [HttpGet("GenerateReward")]
+        public RewardRequest GenerateReward()
+        {
+            // Create reward request with value of 0
+            return new RewardRequest(0);
+        }
+
+        /// <summary>
+        /// Sends reward for event with eventId to the personalizer client
+        /// </summary>
+        /// <param name="eventId"></param>
+        /// <param name="rewardRequest"></param>
+        /// <returns>Status of POST request</returns>
+        [HttpPost("PostReward/{eventId}")]
+        public string PostReward([FromRoute]string eventId, [FromBody]RewardRequest rewardRequest)
+        {
+            try
+            {
+                client.Reward(eventId, rewardRequest);
+            }
+            catch(Exception e)
+            {
+                return e.ToString();
```

Comment:
Looking for a better suggestion of what to return here

Reply:
This should be fine, although `return "OK";' might be better served with the actual response code of `client.Reward`.

---
In reply to: [414823033](https://github.com/Azure-Samples/cognitive-services-personalizer-samples/pull/52#discussion_r414823033) [](ancestors = 414823033)

Reply:
The successful response code is "No content" which I think might be confusing to print on its own. Maybe 204: no content?

Reply:
"204" and something indicating success is good. "OK" just seemed dishonest since that would mean "200".